### ISSUE TITLE
Label encoding for multi-value categoricals

### ISSUE THREAD

**Issue by StanislavVolodarskiy** [deeplearninc/auger-ml#383]
Collect all tags. Let N - number of unique tags. Let M - average number of tags per record (integer). Dustribute all tags in M buckets. Create M columns instead one and distribute tags between new columns according tags-to-bucket distribution. Replace values in columns with hash(set(column_value)). Label encode each column independently.

`def multi_label_encode(c):
    # column is column of lists of strings (tags)
    c = c.str.split(',')

    # fillna actually, fillna cannot accept list as value
    c = c.apply(lambda v: v if isinstance(v, list) else [])

    # make list of all tags
    tags = set()
    count = 0
    for v in c:
        tags.update(v)
        count += max(len(v), 1)
    tags = sorted(tags)

    # split it
    n_splits = int(math.ceil(float(count) / len(c)))
    subsets = map(frozenset, np.array_split(tags, n_splits))
    
    # now column is column of frozensets
    c = c.apply(frozenset)

    # select subsets
    for s in subsets:
        subc = c.apply(s.intersection)
        yield sklearn.preprocessing.LabelEncoder().fit_transform(subc)
`

**Comment by StanislavVolodarskiy** [comment]
One thing missing from suggested solution. fit_transform call should be modified to map empty set into zero. This is to ease sparse representation afterwards.
